neuralop/models/rno.py

```diff
@@ -0,0 +1,164 @@
+import torch
+import numpy as np
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..layers.recurrent_layers import RNO_layer
+from ..layers.padding import DomainPadding
+from ..layers.mlp import MLP
+
+
+class RNO(nn.Module):
+    """
+    N-Dimensional Recurrent Neural Operator.
+    
+    The RNO has an identical architecture to the finite-dimensional GRU, with 
+    the exception that linear matrix-vector multiplications are replaced by  
+    Fourier layers (Li et al., 2021), and for regression problems, the output 
+    nonlinearity is replaced by a SELU activation.
```

Comment:
Can add a short explanation (using the terms h, r etc) and a reference for users to cit